projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -979,11 +984,14 @@ async def _load_copy_types(self, conn: AsyncConnection) -> None:
                     on (k.oid operator(pg_catalog.=) a.attrelid)
                 where n.nspname operator(pg_catalog.=) %s
                 and k.relname operator(pg_catalog.=) %s
-                and a.attname operator(pg_catalog.!=) 'embedding_uuid'
+                AND a.attname = ANY(%s)
                 and a.attnum operator(pg_catalog.>) 0
-                order by a.attnum
             """,
-                (self.vectorizer.target_schema, self.vectorizer.target_table),
+                (
+                    self.vectorizer.target_schema,
+                    self.vectorizer.target_table,
+                    target_columns,
+                ),
             )
             self.copy_types = [row[0] for row in await cursor.fetchall()]
```

Comment:
there is no guarantee right now of the order of the results. The query should return the column name, type and then we can finangle things into the right order